porch/apiserver/pkg/registry/porch/background.go

```diff
@@ -28,14 +28,23 @@ import (
 	"github.com/GoogleContainerTools/kpt/porch/repository/pkg/cache"
 )
 
+func RunBackground(coreClient client.WithWatch, cache *cache.Cache, stopCh <-chan struct{}) {
+	b := background{
+		coreClient: coreClient,
+		cache:      cache,
+	}
+	ctx := context.Background()
+	go b.run(ctx, stopCh)
+}
+
 // background manages background tasks
 type background struct {
 	coreClient client.WithWatch
 	cache      *cache.Cache
 }
 
-// RunForever will run until ctx is done
-func (b *background) RunForever(ctx context.Context) {
+// run will run until ctx is done or stopCh is closed
```

Comment:
Personally, I prefer just putting all my eggs into the context basket.  So at the point where someone passes us a stopCh,I would  immediately translate it to a context.  But that's just me :-)

Reply:
I agree. the stopCh is k8s apiserver concept though that they use to stop the main server so we are caught in the middle there. Looking at the apiserver code, it may be possible to refactor further. I'll see if we can drive the whole apiserver shutdown from context only.